alacritty/src/renderer/text/atlas.rs

```diff
@@ -144,11 +158,24 @@ impl Atlas {
             let (format, buffer) = match &glyph.buffer {
                 BitmapBuffer::Rgb(buffer) => {
                     multicolor = false;
-                    (gl::RGB, buffer)
+                    // Gles context doesn't allow uploading RGB data into RGBA texture, so need
+                    // explicit copy.
+                    if self.is_gles_context {
+                        let mut new_buffer = Vec::with_capacity(buffer.len() / 3 * 4);
+                        for rgb in buffer.chunks_exact(3) {
+                            new_buffer.push(rgb[0]);
+                            new_buffer.push(rgb[1]);
+                            new_buffer.push(rgb[2]);
+                            new_buffer.push(u8::MAX);
+                        }
+                        (gl::RGBA, Cow::Owned(new_buffer))
```

Comment:
Would it make sense to do this at an earlier point, on the `RasterizedGlyph` directly? Just so we can avoid pointlessly moving from one vec to a freshly allocated second one.

Reply:
Do you mean inside the crossfont?

Reply:
Is it a `Vec` or a `&Vec`/`&[]`? I forgot.

Reply:
It's an owning `Vec` that is passed to you from crossfont.

Reply:
Then it should be possible for us to mutate it?

Reply:
Yeah, you need to pass rasterized glyph as mut and such... Be aware that OpenGL 3.3 will also copy under the hood from what I understand.